test/core/experiments/BUILD

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//bazel:grpc_build_system.bzl", "grpc_cc_test", "grpc_cc_library")
+
+grpc_cc_library(
+    name = "test_experiments_lib",
+    srcs = [
+        "test_experiments.cc",
+        "//src/core:lib/experiments/config.cc",
+    ],
+    hdrs = [
+        "test_experiments.h",
+        "//src/core:lib/experiments/config.h",
```

Comment:
We need to define a GRPC_EXPERIMENTS_TEST_FIXTURE macro through the bazel defines. This macro affects experiments/config.cc. It wont be propagated to the config.cc file if we depend on //src/core:experiments - because bazel defines does not propagate the define value to a target's dependencies.

Reply:
Got it. See my other comment on a suggested refactoring. If configs can be injected into the experiments system, this problem goes away.